pkg/deploy/controller/imagechange/controller.go

```diff
@@ -3,161 +3,124 @@ package imagechange
 import (
 	"fmt"
 
+	"k8s.io/kubernetes/pkg/util/sets"
+
 	"github.com/golang/glog"
 
+	"github.com/openshift/origin/pkg/client"
 	deployapi "github.com/openshift/origin/pkg/deploy/api"
 	deployutil "github.com/openshift/origin/pkg/deploy/util"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
-// ImageChangeController increments the version of a DeploymentConfig which has an image
+// ImageChangeController increments the version of a deployment config which has an image
 // change trigger when a tag update to a triggered ImageStream is detected.
 //
 // Use the ImageChangeControllerFactory to create this controller.
 type ImageChangeController struct {
-	deploymentConfigClient deploymentConfigClient
+	listDeploymentConfigs func() ([]*deployapi.DeploymentConfig, error)
```

Comment:
Re: e.g. by having the config change controller also watch image streams
and enqueue related DC

I want to move this to a different controller.  We may want to hold off on
this.